entity-framework/core/testing/index.md

```diff
@@ -1,117 +1,116 @@
+### SQLite as a database fake
 
-Personally, I've never found it a problem having a database service running on my dev machine and I would generally recommend using Developer Edition instead.
-However, LocalDB may be appropriate for some people, especially on less powerful dev machines.
+One possible unit testing approach is to swap your production database (e.g. SQL Server) with SQLite, effectively using it as a testing "fake". Aside from ease of setup, SQLite has a an [in-memory database](https://sqlite.org/inmemorydb.html) feature which is especially useful for unit testing: each test is naturally isolated in its own database, and no actual files need to be managed.
 
-[Running SQL Server](/sql/linux/quickstart-install-connect-docker) (or any other database system) in a Docker container (or similar) is another way to avoid running the database system directly on your development machine.
+However, before doing this, it's important to understand that in EF Core, different database providers behave differently - EF Core does not attempt to abstract every aspect of the underlying database system. Fundamentally, this means that testing against SQLite does not guarantee the same results as against SQL Server, or any other database. Here are some examples of possible behavioral differences:
 
-## Approach 2: SQLite
+* The same LINQ query may return different results on different providers. For example, SQL Server does case-insensitive string comparison by default, whereas SQLite is case-sensitive. This can make your unit tests pass where a test against your actual database would fail (or vice versa).
+* Some queries which work on SQL Server simply aren't supported on SQLite, because the exact SQL support in these two database differs.
+* If your query happens to use a provider-specific method such as SQL Server's [`EF.Functions.DateDiffDay`](xref:core/providers/sql-server/functions#date-and-time-functions), that query will fail on SQLite, and cannot be tested.
+* Raw SQL may work, or it may fail or return different results, depending on exactly what is being done. SQL dialects are different in many ways across databases.
 
-EF Core tests the SQL Server provider primarily by running it against a local SQL Server instance.
-These tests run tens of thousands of queries in a couple of minutes on a fast machine.
-This illustrates that using the real database system can be a performant solution.
-It is a myth that using some lighter-weight database is the only way to run tests quickly.
+Compared to running integration tests against your production database system, it's relatively easy to get started with SQLite, and so many users do. Unfortunately, the above limitations tend to eventually become problematic when unit testing EF Core applications, even if they don't seem to be at the beginning. As a result, we recommend either concentrating on integration tests, or implementing unit tests via the repository pattern as discussed below.
```

Comment:
I disagree with this. I don't think using SQLite is any easier than using your production database system. In fact, it can be harder because a) you need your tests to configure a whole different provider, rather than just a connection string, and b) it is hard to understand and control the lifetime of the SQLite in-memory mode.

Reply:
That's true, though on the other hand:

1. You don't have to set it up (e.g. Docker, PG installation), which some people may consider problematic
2. You don't have to deal with test isolation issues (run tests in transactions, use respawn to reset and reseed database state...).

Note that I don't disagree with you... maybe "it may be easier to get started with SQLite" instead of "it's relatively easy ..."?